src/events.rs

```diff
@@ -154,6 +154,11 @@ impl<T: System> EventsDecoder<T> {
                         self.decode_raw_bytes(&[*arg.clone()], input, output)?
                     }
                 }
+                EventArg::Option(arg) => {
```

Comment:
See the encoding of `Option` here: https://github.com/paritytech/parity-scale-codec/blob/50c50b3fd7c93d9b98b65219208793d4ecb82acd/src/codec.rs#L548.

You would need to check the first byte to see whether it's `Some` or `None`, as in the decode impl: https://github.com/paritytech/parity-scale-codec/blob/50c50b3fd7c93d9b98b65219208793d4ecb82acd/src/codec.rs#L558